pkg/build/api/types.go

```diff
@@ -178,6 +188,26 @@ type GitHubWebHookCause struct {
 	Secret string
 }
 
+// GitLabWebHookCause has information about a GitLab webhook that triggered a
+// build.
+type GitLabWebHookCause struct {
+	// Revision is the git source revision information of the trigger.
+	Revision *SourceRevision
+
+	// Secret is the obfuscated webhook secret that triggered a build.
+	Secret string
+}
+
+// BitbucketWebHookCause has information about a Bitbucket webhook that triggered a
+// build.
+type BitbucketWebHookCause struct {
+	// Revision is the git source revision information of the trigger.
+	Revision *SourceRevision
+
+	// Secret is the obfuscated webhook secret that triggered a build.
+	Secret string
+}
```

Comment:
that's the discussion we're having :)  need @smarterclayton to tell us if we're still ok to change the protobuf or not.  If we are, then yes we want to refactor all the webhookcause structs to use a single common struct field.